### PR TITLE
fixed abrupt transition when fade button clicked

### PR DESCRIPTION
**Pull request by karanimwenda**
`getIntroEndSecond` currently returns `introStart` when `introEnd`  is not set, thus:
`introStart == introEnd`

### CODE REVIEW

File: src/library/autodj/autodjprocessor.cpp [hunk 1]
```diff
@@ -280,7 +280,7 @@ void AutoDJProcessor::fadeNow() {
 
         if (toDeckCurrentSecond >= introStart &&
                 toDeckCurrentSecond <= introEnd &&
-                introEnd >= 0) {
+                introEnd >= 0 && introStart != introEnd) {
```
Comment (Be-ing) [inline src/library/autodj/autodjprocessor.cpp:283]: remove the `introEnd >= 0` as it does not work as intended. Also add a comment explaining why introStart would equal introEnd

Comment (Be-ing) [reply src/library/autodj/autodjprocessor.cpp:283]: explaining that in the commit message is good, but in this case it should be in a code comment so anyone reading the code sees it

Comment (c3n7) [reply src/library/autodj/autodjprocessor.cpp]: I have updated the code and added the comment.

**Comment by Be-ing** [comment]
Thanks, I'm glad we got this fixed. For the future, please add a little bit to the first line of your commit messages to indicate which part of the code you're changing, for example `AutoDJProcessor: remove unneeded condition check in fadeNow`

**Comment by c3n7** [comment]
Noted. My future commit messages will be more informative.

**Pull request merged**